paper-api/src/main/java/org/bukkit/block/CreakingHeart.java

```diff
@@ -1,7 +1,60 @@
 package org.bukkit.block;
 
+import org.bukkit.Location;
+import org.bukkit.entity.Creaking;
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+
 /**
  * Represents a captured state of a creaking heart.
  */
+@NullMarked
 public interface CreakingHeart extends TileState {
+
+    /**
+     * Gets the creaking protecting this creaking heart.
+     *
+     * @return the creaking or null if this creaking heart don't have protector.
+     */
+    @Nullable
+    Creaking getCreaking();
+
+    /**
+     * Sets the creaking protecting this creaking heart.
+     *
+     * @param creaking the creaking or null for make this creaking heart don't have protector
+     * @throws IllegalArgumentException if the creaking passed it's in another world
+     */
+    void setCreaking(@Nullable Creaking creaking);
+
+    /**
+     * Gets the max distance between the Creaking Heart and the Creaking before to remove.
+     *
+     * @return the max distance
+     */
+    int getMaxDistanceForCreaking();
+
+    /**
+     * Sets the max distance between the Creaking Heart and the Creaking before to remove.
+     *
+     * @param distance the max distance
+     * @throws IllegalArgumentException if the distance is negative
+     */
+    void setMaxDistanceForCreaking(int distance);
+
+    /**
+     * Attempts to spawn a creaking for protect this creaking heart.
+     *
+     * @return the {@link Creaking} for protect the creaking heart or null if fails
```

Comment:
I am a bit confused regarding the usage of this method.

Simulating the spawning behaviour of an entity type is something that might be useful API, but I don't think I'd want it to live in random block state types like this.
Developers can very well just spawn a creeking and set it via the setter.
The spawnProtector method really doesn't do more than spawning it + game events.

Reply:
hmmm yeah.. i fell useful just call the same NMS logic but maybe can just remove that and let users use the normal ways.